server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -455,20 +466,39 @@ export function configureWebSocketServices(
 						socket.emit("nack", "", [nackMessage]);
 						return;
 					}
-					contentBatches.forEach((contentBatch) => {
-						const contents = Array.isArray(contentBatch)
-							? contentBatch
-							: [contentBatch];
 
-						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+					if (supportedFeaturesMap.get(clientId)?.submit_signals_v2) {
+						for (const signal of contentBatches) {
+							if (isSentSignalMessage(signal)) {
+								const signalMessage: ISignalMessage = {
+									clientId,
+									...signal,
+								};
+
+								const roomId: string =
+									signal.targetClientId !== undefined
+										? getClientSpecificRoomId(signal.targetClientId)
```

Comment:
what happens if the target client doesn't support Signal v2? Is the client doing something to ensure sending a v2 signal to a v1 client won't be allowed, or is it expected for that to break the v1 client, hopefully causing them to refresh and load a v2 client?

Reply:
The type of the signal being received hasn't changed -- whether it's a v1 or v2 client they receive signals of type ISignalMessage. The difference of v1 and v2 is the type of signal sent to the server.